src/cursor.lua

```diff
@@ -59,7 +59,7 @@ function cursor_methods:count()
 end
 
 function cursor_methods:next()
-  if self.limit_n > 0 and self.i - self.skip_n >= self.limit_n then return nil end
+  if self.limit_n > 0 and self.i >= self.limit_n then return nil end
```

Comment:
To be explicit, we don't want to keep -all- records in memory, we only want to keep this set of records in memory. We return a generator, not a list.